src/CMakeLists.txt

```diff
@@ -69,17 +69,38 @@ if ("${JUPYTER}" STREQUAL "ON")
    add_definitions(-DRB_XCODE)
 endif()
 
+#MESSAGE("Boost location hints:")
+#if (BOOST_ROOT)
+#  MESSAGE("  BOOST_ROOT: ${BOOST_ROOT}")
+#else()
+#  MESSAGE("  BOOST_ROOT not set")
+#endif()
+#if (BOOST_INCLUDEDIR)
+#  MESSAGE("  BOOST_INCLUDEDIR: ${BOOST_INCLUDEDIR}")
+#endif()
+#if (BOOST_LIBRARYDIR)
+#  MESSAGE("  BOOST_LIBRARYDIR: ${BOOST_LIBRARYDIR}")
+#endif()
 MESSAGE("Boost location hints:")
+if (BOOST_ROOT OR BOOST_INCLUDEDIR OR BOOST_LIBRARYDIR) 
+  set(Boost_NO_SYSTEM_PATHS ON)
```

Comment:
What is the benefit of this?  Does it affect linking, or only include paths?

Reply:
Im not 100% sure what this is doing behind the scenes, but I believe that it is ignoring the environment variables that are related to boost (like things that are exported in bashrc).

Reply:
Maybe add a comment line about like # Ignore any environment variable hints about boost paths ?